core/src/main/java/com/sequenceiq/cloudbreak/service/datalake/DiskUpdateService.java

```diff
@@ -0,0 +1,191 @@
+import com.sequenceiq.cloudbreak.util.StackUtil;
+import com.sequenceiq.common.api.type.ResourceType;
+
+@Service
+public class DiskUpdateService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DiskUpdateService.class);
+
+    @Inject
+    private CloudPlatformConnectors cloudPlatformConnectors;
+
+    @Inject
+    private ClusterApiConnectors clusterApiConnectors;
+
+    @Inject
+    private StackUtil stackUtil;
+
+    @Inject
+    private StackDtoService stackDtoService;
+
+    @Inject
+    private CloudParameterCache cloudParameterCache;
+
+    @Inject
+    private StackService stackService;
+
+    @Inject
+    private ResourceService resourceService;
+
+    @Inject
+    private HostOrchestrator hostOrchestrator;
+
+    @Inject
+    private GatewayConfigService gatewayConfigService;
+
+    @Inject
+    private ResourceAttributeUtil resourceAttributeUtil;
+
+    public boolean isDiskTypeChangeSupported(String platform) {
+        return cloudParameterCache.isDiskTypeChangeSupported(platform);
+    }
+
+    public void updateDiskTypeAndSize(DiskUpdateRequest diskUpdateRequest, List<VolumeSetAttributes.Volume> volumesToUpdate,
+            String cloudPlatform) throws Exception {
+        CloudPlatformVariant cloudPlatformVariant = new CloudPlatformVariant(Platform.platform(cloudPlatform), Variant.variant(cloudPlatform));
+        CloudConnector cloudConnector = cloudPlatformConnectors.get(cloudPlatformVariant);
+        StackDto stackDto = stackDtoService.getById(diskUpdateRequest.getStackId());
+        AuthenticatedContext ac = getAuthenticatedContext(cloudConnector, stackDto);
+        List<String> volumeIds = volumesToUpdate.stream().map(VolumeSetAttributes.Volume::getId).toList();
+        cloudConnector.resources().updateDiskVolumes(ac, volumeIds, diskUpdateRequest.getVolumeType(), diskUpdateRequest.getSize());
+    }
+
+    public void stopCluster(long stackId) throws Exception {
+        StackDto stackDto = stackDtoService.getById(stackId);
+        ClusterApi clusterApi = clusterApiConnectors.getConnector(stackDto);
+        clusterApi.clusterModificationService().stopCluster(true);
+    }
+
+    public void startCluster(long stackId) throws Exception {
+        StackDto stackDto = stackDtoService.getById(stackId);
+        ClusterApi clusterApi = clusterApiConnectors.getConnector(stackDto);
+        clusterApi.clusterModificationService().startCluster();
+    }
+
+    public void resizeDisks(long stackId, String instanceGroup) throws Exception {
+        Stack stack = stackService.getByIdWithListsInTransaction(stackId);
+        ResourceType diskResourceType = stack.getDiskResourceType();
+        if (diskResourceType != null) {
+            List<Resource> resourceList = resourceService.findAllByStackIdAndResourceTypeIn(stackId, List.of(diskResourceType))
+                            .stream().filter(res -> null != res.getInstanceId() && res.getResourceType().equals(ResourceType.AWS_VOLUMESET)
```

Comment:
the class name does not indicate that this method is specific for AWS

Reply:
This is not going to be specific for AWS. Until I implement methods for other cloud providers which will be handled in the next quarters - it is just going to be enabled for AWS first

Reply:
but the condition filters for AWS resources